### DISCUSSION TITLE
form_data: Annotated[OAuth2PasswordRequestForm, Depends()]  bugged

### DISCUSSION THREAD

**Discussion by xalteropsx** [fastapi/fastapi#9382] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/token")
async def login(form_data: Annotated[OAuth2PasswordRequestForm, Depends()]):
    user_dict = fake_users_db.get(form_data.username)
    if not user_dict:
        raise HTTPException(status_code=400, detail="Incorrect username or password")
    user = UserInDB(**user_dict)
    hashed_password = fake_hash_password(form_data.password)
    if not hashed_password == user.hashed_password:
        raise HTTPException(status_code=400, detail="Incorrect username or password")

    return {"access_token": user.username, "token_type": "bearer"}
```

### Description

https://fastapi.tiangolo.com/id/tutorial/security/simple-oauth2/#__tabbed_4_2

```
  File "X:\sad\beta\Fastapi\Router\Login.py", line 47, in <module>
    @router.post("/token", response_model=TokenData)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\haide\.virtualenvs\Fastapi-dkQzSXuI\Lib\site-packages\fastapi\routing.py", line 638, in decorator
    self.add_api_route(
  File "C:\Users\haide\.virtualenvs\Fastapi-dkQzSXuI\Lib\site-packages\fastapi\routing.py", line 577, in add_api_route
    route = route_class(
            ^^^^^^^^^^^^
  File "C:\Users\haide\.virtualenvs\Fastapi-dkQzSXuI\Lib\site-packages\fastapi\routing.py", line 446, in __init__
    self.dependant = get_dependant(path=self.path_format, call=self.endpoint)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\haide\.virtualenvs\Fastapi-dkQzSXuI\Lib\site-packages\fastapi\dependencies\utils.py", line 309, in get_dependant
    param_field = get_param_field(
                  ^^^^^^^^^^^^^^^^
  File "C:\Users\haide\.virtualenvs\Fastapi-dkQzSXuI\Lib\site-packages\fastapi\dependencies\utils.py", line 403, in get_param_field
    field = create_response_field(
            ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\haide\.virtualenvs\Fastapi-dkQzSXuI\Lib\site-packages\fastapi\utils.py", line 90, in create_response_field
    raise fastapi.exceptions.FastAPIError(
fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that typing.Annotated[fastapi.security.oauth2.OAuth2PasswordRequestForm, Depends(NoneType)] is a valid Pydantic field type. If you are using a return ty
pe annotation that is not a valid Pydantic field (e.g. Union[Response, dict, None]) you can disable generating the response model from the type annotation with the path operation decorator parameter response_model=None. Read more: https://fastapi.tiangolo.com/tutorial/response-model/
```

### Operating System

Windows

### Operating System Details

window 11

### FastAPI Version

0.89.1

### Python Version

3.11

### Additional Context

always bugged

**Comment by Viicos** [comment, marked as answer]
Upgrade to `0.95.0`

**Comment by Asilias** [comment]
There exists a "non-annotated" version of the code on the tutorial page you linked : [https://fastapi.tiangolo.com/id/tutorial/security/simple-oauth2/#__tabbed_4_2](url)

Try changing all the occurrence of ```var = Annotated[type, Any]``` into ```var: type = Any``` (all occurrences in your code, else it doesn't work)

It worked for me on Centos7, Python 3.6.8 and fastapi==0.83.0
